src/csharp/Microsoft.Spark/Utils/CommandSerDe.cs

```diff
@@ -181,7 +181,8 @@ private sealed class UdfWrapperData
             List<UdfSerDe.UdfData> udfs)
         {
             UdfSerDe.UdfData udfData = UdfSerDe.Serialize(func);
-            if (udfData.MethodName != UdfWrapperMethodName)
+            if (udfData.MethodName != UdfWrapperMethodName ||
+                !Attribute.IsDefined(func.Target.GetType(), typeof(UdfWrapperAttribute)))
```

Comment:
question, do we need this check? looks like all the UDF functions are `Execute', no? Or are you adding this for robustness?

Reply:
If a user defines a udf with the method named `Execute` then this will cause errors.  ie, if we look at `ForeachWriterWrapper.Execute` this is the actual Udf we want to use, and the `ForeachWriterWrapperUdfWrapper.Execute` is wrapping it.

Reply:
Then do we still need this check `udfData.MethodName != UdfWrapperMethodName`?

Reply:
It should be safe to remove that as well.